packages/rebound-runtime/lib/lazy-value.js

```diff
@@ -48,42 +60,59 @@ LazyValue.prototype = {
     return this;
   },
 
-  saveObserver: function(value) {
-    var observers = this.observers;
-    if (!observers) {
-      observers = this.observers = [value];
-    } else {
-      observers.push(value);
+  addObserver: function(path, context) {
```

Comment:
This is where context.js will probably be integrated into. Add a callback on the context to run any relevent __observers on change, add, remove, reset, etc.